### PR TITLE
AAE-10754 Migrate Activiti maven projects to JDK 17

### PR DESCRIPTION
**Pull request by igdianov**
- [x] Update build scripts to use JDK 17
- [x] Update Java version to 17
- [ ] AAE-14392 Integrate Nashorn in Activiti core
- [x] Deprectated preview branch support from CI build replaced by `preview` label

### CODE REVIEW

**Review by erdemedeiros — APPROVED** [review]
LGTM

**Review by atchertchian — APPROVED** [review]
:+1:

File: activiti-core/activiti-engine/src/test/java/org/activiti/engine/test/api/event/ActivityEventsTest.java [hunk 1]
```diff
@@ -486,7 +485,7 @@ public void testActivityTimeOutEvent() {
     Calendar tomorrow = Calendar.getInstance();
     tomorrow.add(Calendar.DAY_OF_YEAR, 1);
     processEngineConfiguration.getClock().setCurrentTime(tomorrow.getTime());
-    waitForJobExecutorToProcessAllJobs(2000, 1000);
+    waitForJobExecutorToProcessAllJobs(5000, 1000);
```
Comment (atchertchian) [inline activiti-core/activiti-engine/src/test/java/org/activiti/engine/test/api/event/ActivityEventsTest.java:488]: do you know why this timeout is impacted please?

**Review by andrea-ligios — APPROVED** [review]
LGTM!

File: pom.xml [hunk 2]
```diff
@@ -563,7 +563,7 @@ limitations under the License.
             <configuration>
               <rules>
                 <requireJavaVersion>
-                  <version>${java.version}</version>
+                  <version>1.${java.version}</version>
```
Comment (atchertchian) [inline pom.xml]: according to my tests this change should not be needed

```suggestion
                  <version>${java.version}</version>
```

**Review by atchertchian — CHANGES_REQUESTED** [review]
please check the change on requireJavaVersion on the enforcer

**Review by atchertchian — APPROVED** [review]

File: activiti-core/activiti-engine/src/test/java/org/activiti/engine/test/api/event/ActivityEventsTest.java (hunk 1 above)
Comment (igdianov) [reply activiti-core/activiti-engine/src/test/java/org/activiti/engine/test/api/event/ActivityEventsTest.java:488]: After JDK 17 update, this test was consistently timing out with 2 seconds...

**Pull request merged**